stdlib/universe/join.go

```diff
@@ -845,11 +841,10 @@ func (c *MergeJoinCache) buildPostJoinSchema() {
 
 // equalJoinKeys compares two keys for equality.
 // Null values are not considered equal when joining (unlike when grouping).
-func equalJoinkeys(left, right flux.GroupKey) bool {
-
-	for j, v := range left.Values() {
+func equalJoinkeys(left, right flux.SortedGroupKey) bool {
+	for j, v := range left.SortedValues() {
 		// value.Equal will return false if both sides are null
-		if !v.Equal(right.Value(j)) {
+		if !v.Equal(right.ValueSorted(j)) {
```

Comment:
Really glad to hear you got to the bottom of this!

I must be missing something though. I see the interface `GroupKey` has a method `Equal` and in the implementation it does a sorted comparison:
https://github.com/influxdata/flux/blob/ac55424f0d5dcf9bc45efba441ede66ed24f2913/internal/execute/groupkey/groupkey.go#L157
Is there a reason that the standard `GroupKey.Equal` would not work?

Reply:
Didn't catch this so I tried it out, and got some failing tests. Looks like the key difference between these two methods is that `v.Equal()` will never consider two nulls to be equal, where as `groupKeyEqual` will.

https://github.com/influxdata/flux/blob/ac55424f0d5dcf9bc45efba441ede66ed24f2913/internal/execute/groupkey/groupkey.go#L171-L174

https://github.com/influxdata/flux/blob/ac55424f0d5dcf9bc45efba441ede66ed24f2913/stdlib/universe/join.go#L842-L844

Reply:
Maybe you could add a method like `EqualNoNulls` to the `GroupKey` interface? (Changing an interface could break downstream repos but I don't think it will.) Both behaviors seem really useful to have.

Reply:
That seems reasonable to me. Working on it.